include/deal.II/fe/fe_tools.templates.h

```diff
@@ -187,15 +187,21 @@ namespace FETools
           block_indices.push_back(fes[base]->n_dofs_per_cell());
 
       const ReferenceCell reference_cell = fes.front()->reference_cell();
-      Assert(std::all_of(fes.begin(),
-                         fes.end(),
-                         [reference_cell](
-                           const FiniteElement<dim, spacedim> *fe) {
-                           return fe->reference_cell() == reference_cell;
-                         }),
+#ifdef DEBUG
+      // Workaround: in release mode we use all arguments in the Assert macro
+      // in an unevaluated context. Using a lambda in an unevaluated context is
+      // only available beginning with C++20.
+      const bool reference_cells_consistent =
+        std::all_of(fes.begin(),
+                    fes.end(),
+                    [reference_cell](const FiniteElement<dim, spacedim> *fe) {
+                      return fe->reference_cell() == reference_cell;
+                    });
+      Assert(reference_cells_consistent,
```

Comment:
The comment here is only understandable if you know the history of the code, because it is not clear at all what you are working around. Same in all of the other places. You should start by saying what you are commenting on -- namely, that you are using a named variable rather than putting the expression right into the assertion.

Reply:
@bangerth We can alternatively simply switch to c++20 or back out the changes in the `Assert` macro and undo all `[[maybe_unused]]` changes.

Reply:
Discussed and adressed.